src/test/regress/sql/tableam.sql

```diff
@@ -0,0 +1,143 @@
+select * from test_ref;
+insert into test_ref values (1);
+
+-- we should error on following, since this AM is append only
+SET client_min_messages TO ERROR;
+delete from test_ref;
+update test_ref set a=2;
+RESET client_min_messages;
+
+-- ddl events should include "USING fake_am"
+SELECT * FROM master_get_table_ddl_events('test_ref');
+
+-- replicate to coordinator
+SET client_min_messages TO WARNING;
+SELECT 1 FROM master_add_node('localhost', :master_port, groupid => 0);
+RESET client_min_messages;
+delete from test_ref;
+SELECT master_remove_node('localhost', :master_port);
+
+--
+-- Range partitioned table using a non-default table access method
+--
+
+CREATE TABLE test_range_dist(id int, val int) using fake_am;
+SELECT create_distributed_table('test_range_dist', 'id', 'range');
+CALL public.create_range_partitioned_shards('test_range_dist', '{"0","25"}','{"24","49"}');
+
+select * from test_range_dist;
+insert into test_range_dist values (1, 1);
+COPY test_range_dist FROM PROGRAM 'echo 0, 0 && echo 1, -1 && echo 2, 4 && echo 3, 9' WITH CSV;
+COPY test_range_dist FROM PROGRAM 'echo 25, 16 && echo 26, 1 && echo 27, 4 && echo 7, 9' WITH CSV;
+
+
+-- ddl events should include "USING fake_am"
+SELECT * FROM master_get_table_ddl_events('test_range_dist');
+
+--
+-- Test master_copy_shard_placement with a fake_am table
+--
+
+select a.shardid, a.nodeport
+FROM pg_dist_shard b, pg_dist_shard_placement a
+WHERE a.shardid=b.shardid AND logicalrelid = 'test_range_dist'::regclass::oid
+ORDER BY a.shardid, nodeport;
+
+SELECT master_copy_shard_placement(
+           get_shard_id_for_distribution_column('test_range_dist', '1'),
+           'localhost', :worker_1_port,
+           'localhost', :worker_2_port,
+           do_repair := false);
+
+select a.shardid, a.nodeport
+FROM pg_dist_shard b, pg_dist_shard_placement a
+WHERE a.shardid=b.shardid AND logicalrelid = 'test_range_dist'::regclass::oid
+ORDER BY a.shardid, nodeport;
+
+-- verify that data was copied correctly
+
+\c - - - :worker_1_port
+select * from test_tableam.test_range_dist_60005 ORDER BY id;
+
+\c - - - :worker_2_port
+select * from test_tableam.test_range_dist_60005 ORDER BY id;
+
+\c - - - :master_port
+
+SET search_path TO test_tableam;
+
+--
+-- Append partitioned table using a non-default table access method
+--
+
+CREATE TABLE test_append_dist(id int, val int) using fake_am;
+SELECT create_distributed_table('test_append_dist', 'id', 'append');
+
+COPY test_append_dist FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+COPY test_append_dist FROM PROGRAM 'echo 4, 16 && echo 5, 1 && echo 6, 4 && echo 7, 9' WITH CSV;
+
+SELECT shardminvalue, shardmaxvalue FROM pg_dist_shard
+WHERE logicalrelid='test_append_dist'::regclass::oid
+ORDER BY shardminvalue, shardmaxvalue;
+
+-- ddl events should include "USING fake_am"
+SELECT * FROM master_get_table_ddl_events('test_append_dist');
+
+\set VERBOSITY terse
+drop schema test_tableam cascade;
+\set VERBOSITY default
+
```

Comment:
maybe on thing missing is a partitioned table test

Reply:
This mostly makes sense for the different dependencies which parent tables and its partitions can have, which was a bug that I solved by moving the object propagation to `CreateDistributedTable`. This might be hard though given we incorporate `fake_am` in citus instead of adding the blackhole am to our testing suite.